### PR TITLE
Add support for java-cfenv

### PR DESCRIPTION
**Pull request by eddumelendez**
Fixes gh-1479

### CODE REVIEW

File: spring-cloud-gcp-autoconfigure/pom.xml [hunk 1]
```diff
@@ -117,6 +117,18 @@
             <scope>test</scope>
         </dependency>
 
+        <!-- Java-CFEnv -->
+        <dependency>
+            <groupId>io.pivotal.cfenv</groupId>
+            <artifactId>java-cfenv</artifactId>
```
Comment (meltsufin) [inline spring-cloud-gcp-autoconfigure/pom.xml]: This should be optional, and we should document that one must add this dependency if they're using CF, or maybe we need to create a starter for CF.

**Comment by meltsufin** [comment]
@artembilan I think it's a little odd to add this non-core-spring dependency for everyone using the project, even if they're not using CF. Should we instead create a tiny CF starter for it? I think that would also advertise support for CF better because we would list it as one of the starters provided by the project. WDYT?

**Comment by artembilan** [comment]
>  Should we instead create a tiny CF starter for it?

Well, let's ask @eddumelendez then!
I think he has some experience with it in other projects since he has come to our side for the fix.

**Comment by eddumelendez** [comment]
I think create a starter for it make sense since starters always provide the right set of dependencies. Also, since no every `spring-cloud-gcp` project is over `cloudfoundry` this will bring an additional dependency which is not needed. For that reasons, I will be in favor of a new starter.

If you both agreed I can work on it in this PR. WDYT?

**Comment by artembilan** [comment]
Good. Then Docs pointer would be great addition to explain what is need when in CF.
Thanks

**Comment by meltsufin** [comment]
Since the creation of the CF module seems a little bit separate from the intents of this PR, how about we merge this one, and then work on the new module in a follow-up PR?

**Comment by eddumelendez** [comment]
ok, cool! I will send another PR then

**Pull request merged**